commands2/ramsetecommand.py

```diff
@@ -0,0 +1,229 @@
+        self.usePID = False
+        # All the parameter checks pass, inform scheduler.  Type ignoring is set explictitly for the linter because this
+        # implementation declares the tuple explicitly, whereas the general implementations use the unpack operator (*)
+        # to pass the requirements to the scheduler.
+        self.addRequirements(requirements)  # type: ignore
+
+    @overload
+    def __init__(
+        self,
+        trajectory: Trajectory,
+        pose: Callable[[], Pose2d],
+        controller: RamseteController,
+        kinematics: DifferentialDriveKinematics,
+        requirements: Tuple[Subsystem],
+        output: Callable[[float, float], None],
+        feedforward: SimpleMotorFeedforwardMeters,
+        leftController: PIDController,
+        rightController: PIDController,
+        wheelSpeeds: Callable[[], DifferentialDriveWheelSpeeds],
+    ):
+        """Constructs a new RamseteCommand that, when executed, will follow the provided trajectory. PID
+        control and feedforward are handled internally, and outputs are scaled -12 to 12 representing
+        units of volts.
+
+        Note: The controller will *not* set the outputVolts to zero upon completion of the path -
+        this is left to the user, since it is not appropriate for paths with nonstationary endstates.
+
+        :param trajectory:      The trajectory to follow.
+        :param pose:            A function that supplies the robot pose - use one of the odometry classes to
+                                provide this.
+        :param controller:      The RAMSETE controller used to follow the trajectory.
+        :param kinematics:      The kinematics for the robot drivetrain.
+        :param requirements:    The subsystems to require.
+        :param output      A function that consumes the computed left and right outputs in `units.volts`
+        :param feedforward      The feedforward to use for the drive
+        :param leftController:  The PIDController for the left side of the robot drive.
+        :param rightController: The PIDController for the right side of the robot drive.
+        :param wheelSpeeds:     A function that supplies the speeds of the left and right sides of the robot
+                                drive.
+        """
+        super().__init__()
+
+        self._timer = Timer()
+
+        # Store all the requirements that are required
+        self.trajectory = trajectory
+        self.pose = pose
+        self.follower = controller
+        self.kinematics = kinematics
+        self.output = output
+        self.leftController = leftController
+        self.rightController = rightController
+        self.wheelspeeds = wheelSpeeds
+        self.feedforward = feedforward
+        self._prevSpeeds = DifferentialDriveWheelSpeeds()
+        self._prevTime = -1.0
+        self.usePID = True
+        # All the parameter checks pass, inform scheduler.  Type ignoring is set explictitly for the linter because this
```

Comment:
When I keep things as they are with other commands, the '*' throws off the dispatch to the point where incorrect constructors get called, as well as mypy coming up with type errors.  Basically, for me to get it to work, I had to declare the requirements explicitly as a Tuple, without unpacking since that's what addRequirements wants.  Mypy was not happy with this solution, so I just type:ignore.

There is possibly a way to get it to work, but I couldn't figure it out.